delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/resources/ResourcesImpl.scala

```diff
@@ -82,6 +82,22 @@ final class ResourcesImpl private (
     } yield res
   }.span("updateResource")
 
+  override def refresh(
+      id: IdSegment,
+      projectRef: ProjectRef,
+      schemaOpt: Option[IdSegment]
+  )(implicit caller: Caller): IO[ResourceRejection, DataResource] = {
+    for {
+      projectContext        <- fetchContext.onModify(projectRef)
+      iri                   <- expandIri(id, projectContext)
+      schemeRefOpt          <- expandResourceRef(schemaOpt, projectContext)
+      (source, rev)         <- fetch(IdSegmentRef(iri), projectRef, schemaOpt).map(r => r.value.source -> r.rev)
```

Comment:
This is where we are getting the revision from. Not sure about this

Reply:
I think we should use:
```scala
 log.stateOr(projectRef, iri, notFound)
```